internal/mode/static/manager.go

```diff
@@ -224,6 +231,13 @@ func StartManager(cfg config.Config) error {
 		return fmt.Errorf("cannot register event loop: %w", err)
 	}
 
+	if cfg.MetricsConfig.MetricsEnabled {
+		err = configureNginxMetrics(cfg.GatewayClassName)
```

Comment:
got this error. NKG started before NGINX:
```
Defaulted container "nginx-gateway" out of: nginx-gateway, nginx
{"level":"info","ts":"2023-08-24T21:38:08Z","msg":"Starting NGINX Kubernetes Gateway in static mode","version":"edge","commit":"5602c2482a870de128ead23580d1d5d0eae8111a","date":"2023-08-24T21:29:13Z"}
Error: failed to start control loop: cannot register nginx metrics: cannot get NGINX metrics: failed to get http://config-status/stub_status: Get "http://config-status/stub_status": dial unix /var/run/nginx/nginx-status.sock: connect: no such file or directory
Usage:
  gateway static-mode [flags]

Flags:
  -c, --config string                The name of the NginxGateway resource to be used for this controller's dynamic configuration. Lives in the same Namespace as the controller. (default "")
      --enable-metrics               Enable exposing metrics in the Prometheus format.
      --gateway string               The namespaced name of the Gateway resource to use. Must be of the form: NAMESPACE/NAME. If not specified, the control plane will process all Gateways for the configured GatewayClass. However, among them, it will choose the oldest resource by creation timestamp. If the timestamps are equal, it will choose the resource that appears first in alphabetical order by {namespace}/{name}.
  -h, --help                         help for static-mode
      --metrics-listen-port int      Set the port where the metrics are exposed. [1023 - 65535] (default 9113)
      --metrics-secure-serving       Enables serving metrics via https. By default metrics are served via http.
      --update-gatewayclass-status   Update the status of the GatewayClass resource. (default true)

Global Flags:
      --gateway-ctlr-name string   The name of the Gateway controller. The controller name must be of the form: DOMAIN/PATH. The controller's domain is 'gateway.nginx.org' (default "")
      --gatewayclass string        The name of the GatewayClass resource. Every NGINX Gateway must have a unique corresponding GatewayClass resource. (default "")

failed to start control loop: cannot register nginx metrics: cannot get NGINX metrics: failed to get http://config-status/stub_status: Get "http://config-status/stub_status": dial unix /var/run/nginx/nginx-status.sock: connect: no such file or directory
```
that lead to a NKG restart. Is it possible to avoid that?

Reply:
Good catch! I added a check which polls for the availability of the URL (similar to our check for the pid file) before instantiating the collector (unit test pending 😄). Let me know what you think!